transport/tchannel/config.go

```diff
@@ -75,8 +77,53 @@ type InboundTLSConfig struct {
 // 	  myservice:
 // 	    tchannel:
 // 	      peer: 127.0.0.1:4040
+//
 type OutboundConfig struct {
 	yarpcconfig.PeerChooser
+	// TLS config enables TLS outbound.
+	//
+	//  tchannel:
+	//    peer: 127.0.0.1:4040
+	//    tls:
+	//      mode: enforced
+	//      spiffe-ids:
+	//        - destination-id
+	TLS OutboundTLSConfig `config:"tls"`
+}
+
+// OutboundTLSConfig configures TLS for a TChannel outbound.
+type OutboundTLSConfig struct {
+	// Mode when set to Enforced enables TLS outbound and
+	// outbound TLS configuration provider will be used for fetching tls.Config.
+	Mode yarpctls.Mode `config:"mode,interpolate"`
+	// SpiffeIDs is a list of the accepted server spiffe IDs.
+	SpiffeIDs []string `config:"spiffe-ids"`
+}
+
+// getPeerTransport returns peer transport to be used in peer chooser creation.
+func (c OutboundConfig) getPeerTransport(transport *Transport, destName string) (peer.Transport, error) {
+	if c.TLS.Mode == yarpctls.Disabled {
+		return transport, nil
+	}
+
+	if c.TLS.Mode == yarpctls.Permissive {
+		return nil, errors.New("outbound does not support permissive TLS mode")
```

Comment:
nit: all those errors (two more down) seems to be constant. lets move to a top `var` section?

Reply:
Since these errors are exposed publically, I thought of keeping them within the error scope similar to stay inline with other errors returned from this pkg:
https://github.com/yarpc/yarpc-go/blob/792f032c626787c98d49051aac80d6fb44fb56c0/transport/tchannel/config.go#L131-L139

wdyt?

Reply:
sure consistency is good. Can always refactor later. Also this is on startup time, so it is fine to allocate.